crest/Assets/Crest/Crest/Scripts/Helpers/UnderwaterPostProcess.cs

```diff
@@ -0,0 +1,244 @@
+        }
+
+        void OnRenderImage(RenderTexture source, RenderTexture target)
+        {
+            if (_commandBuffer == null)
+            {
+                _commandBuffer = new CommandBuffer();
+                _commandBuffer.name = "Underwater Post Process";
+            }
+
+            bool definitelyAboveTheWater = false;
+            {
+                float oceanHeight = OceanRenderer.Instance.transform.position.y;
+                float maxOceanVerticalDisplacement = OceanRenderer.Instance.MaxVertDisplacement * 0.5f;
+                float cameraHeight = _mainCamera.transform.position.y;
+                definitelyAboveTheWater = (cameraHeight - maxOceanVerticalDisplacement) >= oceanHeight;
+            }
+
+            if (GL.wireframe || definitelyAboveTheWater)
+            {
+                Graphics.Blit(source, target);
+                _oceanChunksToRender.Clear();
+                return;
+            }
+
+            if (_textureMask == null)
+            {
+                _textureMask = new RenderTexture(source);
+                _textureMask.name = "Ocean Mask";
+                // @Memory: We could investigate making this an 8-bit texture
+                // instead to reduce GPU memory usage.
+                _textureMask.format = RenderTextureFormat.RHalf;
+                _textureMask.Create();
+
+                _depthBuffer = new RenderTexture(source);
+                _depthBuffer.name = "Ocean Mask Depth";
+                _depthBuffer.format = RenderTextureFormat.Depth;
+                _depthBuffer.Create();
+            }
+
+            // Get all ocean chunks and render them using cmd buffer, but with mask shader
+            _commandBuffer.SetRenderTarget(_textureMask.colorBuffer, _depthBuffer.depthBuffer);
+            _commandBuffer.ClearRenderTarget(true, true, Color.white * UNDERWATER_MASK_NO_MASK);
+            _commandBuffer.SetViewProjectionMatrices(_mainCamera.worldToCameraMatrix, _mainCamera.projectionMatrix);
+
+            foreach (var chunk in _oceanChunksToRender)
+            {
+                _commandBuffer.DrawRenderer(chunk, _oceanMaskMaterial);
+            }
+
+            {
+                // The same camera has to perform post-processing when performing
+                // VR multipass rendering so don't clear chunks until we have
+                // renderered the right eye.
+                // This was the approach recommended by Unity's post-processing
+                // lead Thomas Hourdel.
+                bool saveChunksToRender =
+                    XRSettings.enabled &&
+                    XRSettings.stereoRenderingMode == XRSettings.StereoRenderingMode.MultiPass &&
+                    _mainCamera.stereoActiveEye != Camera.MonoOrStereoscopicEye.Right;
+
+                if (!saveChunksToRender) _oceanChunksToRender.Clear();
+            }
+
+            _underwaterPostProcessMaterial.CopyPropertiesFromMaterial(OceanRenderer.Instance.OceanMaterial);
```

Comment:
I'm unsure about calling this every frame. in the old implementation there was a checkbox on the script to copy the params every frame. it would default to on so be friendly by default, but allows people to turn it off if they know they don't need it. would that be an option here or does it need to happen every frame?

Reply:
That was me going for the "safest" option with regards to keeping the two materials in-sync. I'm happy following your lead on this here. I guess we could have a checkbox to enable copying on every frame, and also add a button to copy the material params in the editor and expose a function to drive that from code?

Reply:
I'm happy to implement this if you think that these would be good approaches.

Reply:
I tried to make it copy the params once on start, and then make the subsequent copies optional. Unfortunately it doenst work - the underwater appears black.

My best guess is that one of our ocean shader uniforms does not get set before init() time, and this value is therefore not set on the underwater material. This is a shame but i guess we'll leave it as is.

Reply:
I've pushed a slightly different arrangement that essentially does this